### PR TITLE
Add FP32 acc mode to Bmm

### PR DESCRIPTION
**Pull request by yugaoTT**
1. add emit entry for fp32_acc_mode for hit compile
3. add pybindings for fp32_acc_mode and math_approx_mode
4. add kernel support (add template args for fp32_acc_mode)

### CODE REVIEW

**Comment by davorchap** [comment]
@yugaoTT do we ensure that subblock for DST is <= 4 tiles when using FP32 accum?

**Comment by razorback3** [comment]
What exactly does _math_approx_mode_ do?
I understand that giving low math fidelity gives better computation performance with less accuracy, but how is the relationship with _math_approx_mode_?

**Comment by yugaoTT** [comment]
> @yugaoTT do we ensure that subblock for DST is <= 4 tiles when using FP32 accum?

@davorchap I can add `tt-asserts` to the bmm host code

File: tt_metal/jit_build/genfiles.cpp [hunk 1]
```diff
@@ -329,9 +345,11 @@ void jit_build_genfiles_descriptors(const JitBuildEnv& env,
         std::thread td( [&]() { generate_data_format_descriptors(options, env.get_arch()); } );
         std::thread tm( [&]() { generate_math_fidelity_descriptor(options); } );
         std::thread ta( [&]() { generate_math_approx_mode_descriptor(options); } );
+        std::thread tf( [&]() { generate_dst_accum_mode_descriptor(options); } );
```
[the remark this answers is not in the document]
Comment (yugaoTT) [reply tt_metal/jit_build/genfiles.cpp:348]: Yes I can move the new function to math approx mode. I think this file should be always needed, as it is required by the llks, previously we just set this fp32 variable to false.

[the remark this answers is not in the document]
Comment (yugaoTT) [reply tt_metal/jit_build/genfiles.cpp:348]: on second thought, since this file is always needed, should we just keep it in a separate thread?

**Pull request merged**